route/build/src/water_balance.f90

```diff
@@ -111,6 +112,80 @@ SUBROUTINE comp_reach_wb(seg_id,     &     ! input: reach/lake id
 
   END SUBROUTINE comp_reach_wb
 
+  ! *********************************************************************
+  ! public subroutine: compute reach water balance
+  ! *********************************************************************
+  SUBROUTINE comp_reach_mb(seg_id,     &     ! input: reach/lake id
+                           ixRoute,    &     ! input: index of routing method
+                           Cupstream,  &     ! input: mass flux from upstream
+                           Clat,       &     ! input: lateral flow into reach
+                           RCHFLX_in,  &     ! inout: reach flux data structure
+                           verbose,    &     !
+                           lakeFlag,   &     !
+                           tolerance)
+  ! Descriptions
+  ! Compute constituent mass balance per simulation time step and reach/lake
+  ! During a time step dt = t1 - t0 [sec]
+  ! dMass    [mg]   = mass(t1) - mass(t0)
+  ! flux_in  [mg/s] = inflow(dt) + lateral(dt)
+  ! flux_out [mg/s] = outflow(dt)
+  !
+  ! Compare dMass vs (flux_in - flux_out)*dt
+  implicit none
+  ! Argument variables:
+  integer(i4b), intent(in)                 :: seg_id         ! input: routing method index
+  integer(i4b), intent(in)                 :: ixRoute        ! input: routing method index
+  real(dp),     intent(in)                 :: Cupstream      ! input: total inflow from upstream reaches
+  real(dp),     intent(in)                 :: Clat           ! input: lateral flow into reach
+  type(STRFLX), intent(inout)              :: RCHFLX_in      ! inout: Reach fluxes data structure
+  logical(lgt), intent(in)                 :: lakeFlag       ! input: reach index to be examined
+  logical(lgt), intent(in)                 :: verbose        ! input: reach index to be examined
+  real(dp),     optional, intent(in)       :: tolerance      ! input: wb error tolerance trigering print out
+  ! Local variables:
+  real(dp)                                 :: MBerr          ! Mass balance error [mg]
+  real(dp)                                 :: mb_tol         ! mass balance error tolerance [mg]
+  real(dp)                                 :: dMass          ! mass change per time step [mg]
+  real(dp)                                 :: Cin            ! mass flux from upstream to a reach [mg]
+  real(dp)                                 :: Clateral       ! lateral mass flux [mg]
+  real(dp)                                 :: Cout           ! mass flux out of a reach [mg]
+
+  if (present(tolerance)) then
+    mb_tol=tolerance
+  else
+    mb_tol=2.e-5_dp
```

Comment:
What is the significance of this specific value here?

Reply:
2.e-5 seems pretty permissive. Is this usually overridden with tolerance sent in?

Reply:
actually mass balance error is somewhere between 1.e-5 and 1.e-4 in mg at max.  The biggest error happens around DOC mass in river is order of 1.e+6 mg, or even greater. 
How CTSM mass balance tolerance is determined??

Reply:
OK, from our discussion this seems fine. Units of mg makes this make sense to me. A larger unit like Tons, or 1000kg it would need a smaller value.

But, you've tested to see what the error is acting like and it seems reasonable when this is a small unit. So resolving.